slides/src/content/services.md

```diff
@@ -114,4 +114,137 @@ import { HttpModule } from '@angular/http';
+
+---
+
+## Handling `http` rejections
+
+To catch rejections we can use the `subscribe` operator's `error` and `complete` callbacks
+
+```ts
+  this.http.post(`${ BASE_URL }/auth/login`, payload)
+    .map(response => response.json())
+    .subscribe(
+      (authData) => this.storeToken(authData.id_token), // Next function
+      (err) => console.error(err),                      // Error function
+      () => console.log('Authentication Complete')      // Complete function
+    );
+```
+
+---
+
+## Using the `.catch` operator
+
+- The `.catch` operator allows us to catch errors on a stream, do something, then pass the exception onwards
+- This also means we can inspect the error and decide how to handle it
+
+```ts
+  return this.http.get('https://api.spotify.com/v1/dsds?q=' + term + '&type=artist')
+    .map((response) => response.json())
+    .catch(err => {
+      if (err.status >==  500) {
+        return cachedVersion();
+      } else {
+        return Observable.throw(
+          new Error(`${ err.status } ${ err.statusText }`)
+        );
+      }
+    });
+```
+[View Example](http://plnkr.co/edit/3lCaeI?p=preview)
+
+---
+
+## Cancel a Request
+
+Cancelling an HTTP request is a common requirement. If you have a queue of requests and a new
+request supersedes the pending requests, you can call `unsubscribe` on the pending request's subscription.
+
+```ts
+  const request = this.searchService.search(this.searchField.value)
+    .subscribe(
+      result => { this.result = result.artists.items; },
+      err => { this.errorMessage = err.message; },
+      () => { console.log('Completed'); }
+    );
+
+  request.unsubscribe();
+```
+[View Example](http://plnkr.co/edit/XQL8v9?p=preview)
+
+---
+
+## Retry a request
+
+- It is possible to retry a failed request using the `.retry` operator when something goes wrong, for example, the user's connection may dropped.
+- `retry` takes a `retryCount` argument that specifies the number of times to retry a failed request.
+- If you do not specify `retryCount`, the request will be retried indefinitely.
+
+```ts
+  search(term: string) {
+    let tryCount = 0;
+    return this.http.get('https://api.spotify.com/v1/dsds?q=' + term + '&type=artist')
+      .map(response => response.json())
+      .retry(3);  // Will retry failed request 3 times
+}
+```
+- **Note:** The `onError` callback will not execute during the retry phase. The stream will only throw an error after the retry phase is complete
+
+[View Example](http://plnkr.co/edit/zSAWwV?p=preview)
+
+---
+
+## Search with `flatMap`
+
+Let's say we wanted to add a search feature that makes an http request and updates search results with every keypress event in a text field.
+By creating an `Observable` subscribed to events from the input field, and on every change to the input value we perform the HTTP request, which
+is also an `Observable`, we create an `Observable` of an `Observable`.
```

Comment:
I feel like making the descriptions for `flatMap` and `switchMap` more concise would include removing the context of using it with HTTP requests. At which point they might as well be a part of the proposed RxJs section. Thoughts?